lib/train/transports/clients/azure/vault.rb

```diff
@@ -0,0 +1,36 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+require 'azure_mgmt_key_vault'
+
+# Wrapper class for ::Azure::KeyVault::Profiles::Latest::Mgmt::Client allowing custom configuration,
+# for example, defining additional settings for the ::MsRestAzure::ApplicationTokenProvider.
+class Vault
+  def initialize(vault_name)
+    if vault_name.nil?
+      raise ::Train::UserError, 'Vault Name cannot be nil'
+    end
+    @auth_endpoint = MsRestAzure::AzureEnvironments::AzureCloud.active_directory_endpoint_url
+    @api_endpoint = "https://#{vault_name}#{MsRestAzure::AzureEnvironments::AzureCloud.key_vault_dns_suffix}"
```

Comment:
Is this missing a separator between the interpolated values? A period?

Reply:
the suffix adds on the leading `.` :)